sos/cleaner/mappings/hostname_map.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import re
+
+from sos.cleaner.mappings import SoSMap
+
+
+class SoSHostnameMap(SoSMap):
+    """Mapping store for hostnames and domain names
+
+    Hostnames are obfuscated using an incrementing counter based on the total
+    number of hosts matched regardless of domain name.
+
+    Domain names are obfuscated based on the host's hostname, plus any user
+    defined domains passed in by the `--domains` options.
+
+    Domains are obfuscated as whole units, meaning the domains 'example.com'
+    and 'host.foo.example.com' will be separately obfuscated with no relation
+    for example as 'obfuscatedomdain1.com' and 'obfuscatedomain2.com'.
+
+    Top-level domains are left untouched.
+    """
+
+    ignore_matches = [
+        'localhost',
+        '.*localdomain.*',
+        'com..*'
+    ]
+
+    host_count = 0
+    domain_count = 0
+    _domains = {}
+    hosts = {}
+
+    def __init__(self, opt_domains):
+        super(SoSHostnameMap, self).__init__()
+        self.load_domains_from_options(opt_domains)
+
+    def load_domains_from_options(self, domains):
+        for domain in domains:
+            self.sanitize_domain(domain.split('.'))
+
+    def domain_name_in_loaded_domains(self, domain):
+        """Check if a potential domain is in one of the domains we've loaded
+        and should be obfuscated
+        """
+        host = domain.split('.')
+        if len(host) < 2:
+            return False
+        else:
+            domain = host[0:-1]
+            for known_domain in self._domains:
+                if known_domain in domain:
+                    return True
+        return False
+
+    def get(self, item):
+        if item.startswith(('.', '_')):
+            item = item.lstrip('._')
+        item = item.strip()
+        if not self.domain_name_in_loaded_domains(item):
+            return item
```

Comment:
I dont understand here: why do we return `item` (i.e. the not-obfuscated key) when the `item` was evaluated as potential domain we should obfuscated (cf. description of `domain_name_in_loaded_domains`)?